src/bank/parties/Customer.java

```diff
@@ -0,0 +1,140 @@
+ * @author Kobby Asante-Ansong
+ */
+public class Customer implements PayingParty {
+    
+    /** The primary checking account associated with the customer. */
+    private BankAccount checking = new BankAccount(0, "checking", null);
+
+    /** List of savings accounts associated with the customer. */
+    private List<BankAccount> savingsAccounts = new ArrayList<>();
+
+    /** List of credit accounts associated with the customer. */
+    private List<CreditAccount> creditAccounts = new ArrayList<>();
+
+    /** A map to track the original balances of all accounts for reset purposes. */
+    private Map<Account, Double> balanceTracker = new HashMap<>();
+
+
+    /**
+     * Constructs a Customer with an initial checking account.
+     *
+     * @param checking The initial checking account associated with the customer.
+     */
+    public Customer(BankAccount checking){
+        this.checking = checking;
+    }
+
+    /**
+     * Initiates a payment transaction by deducting the specified amount from the
+     * customer's accounts. Handles insufficient funds using the CheckProcessor and
+     * an InsufficientFundsException.
+     *
+     * @param amount The amount to be paid.
+     */
+    public void pay(double amount){
+        storeBalances();
+        if(CheckProcessor.processCheck(checking, amount)){
+            System.out.println("SUCCESS!!");
+            return;
+        }else{
+            Account previousAccount = checking;
+            for(BankAccount account: savingsAccounts){
+                if(previousAccount.balance > 0){
+                    amount -= previousAccount.balance;
+                    previousAccount.balance = 0;
+                }
+                if(CheckProcessor.processCheck(account, amount)){
+                    System.out.println("SUCCESS!!");
+                    return;
+                }
+                previousAccount = account;
+            }
+            for(Account account: creditAccounts){
+                if(previousAccount instanceof BankAccount){
+                    if(previousAccount.balance > 0){
+                        amount -= previousAccount.balance;
+                        previousAccount.balance = 0;
+                    }
+                }else{
+                    if(((CreditAccount) previousAccount).limitLeft > 0){
+                        amount -= ((CreditAccount) previousAccount).limitLeft;
+                        ((CreditAccount) previousAccount).limitLeft = 0;
+                        ((CreditAccount) previousAccount).balance = ((CreditAccount) previousAccount).limit;
+                    }
+                }
+                try{
+                    account.deduct(amount);
+                    System.out.println("SUCCESS!!");
+                    return;
+                }catch (InsufficientFundsException e){
+                    System.out.println(e.getMessage());
+                    previousAccount = account;
+                }
+            }
```

Comment:
Rather than for loops for CoR pattern this needs to only call it on the first link in the chain and then that class passes it down the chain if it can't satisfy it itself.